devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java

```diff
@@ -115,12 +115,7 @@ protected void doExecute() throws MojoExecutionException {
                 }
             }
 
-            // Essentially what this does is to enable the native package type even if a different package type is set
-            // in application properties. This is done to preserve what users expect to happen when
-            // they execute "mvn package -Dnative" even if quarkus.package.type has been set in application.properties
-            if (setPackageTypeSystemPropertyIfNativeProfileEnabled()) {
-                propertiesToClear.add(PACKAGE_TYPE_PROP);
```

Comment:
It's brittle but we need to make sure we clear the properties we set, since otherwise they will remain set for subsequent goal executions.